mesop/web/src/component_renderer/component_renderer.ts

```diff
@@ -233,6 +216,34 @@ export class ComponentRenderer {
     }
   }
 
+  private updateCustomElementChildren() {
+    const existingChildren = Array.from(
+      this.customElement!.querySelectorAll(COMPONENT_RENDERER_ELEMENT_NAME),
+    );
+    const newChildren = this.component.getChildrenList();
+
+    // Update or add children
+    for (let i = 0; i < newChildren.length; i++) {
+      const child = newChildren[i];
+      if (i < existingChildren.length) {
+        // Update existing child
+        (existingChildren[i] as any)['component'] = child;
```

Comment:
Are there any edge cases where this could have unexpected behavior?

For example, if the order of the components got swapped, will Angular still know the text input is the same text input and thus will not re-render it?

Let's say we start with:

text1
text2
button1

And after the state update we get:

text2
button1
text1

And now let's say we get this update:

button3
button2
text1
button1
text2

I think the components will render, but not sure if any of the text inputs would get reset in this scenario.

Reply:
Discussed offline: this isn't unique to web components and is a longstanding issue: #312